### PR TITLE
[sc-20143] Create BambooHR Catalog

### PR DESCRIPTION
**Pull request by arooxa**
This catalog is for the BambooHR adapter which currently only supports the Employee entity.

Note: This catalog was testing successfully in dev

### CODE REVIEW

**Comment by r-rakshith** [comment]
Let's reopen this when you pick up this story

**Review by r-rakshith — CHANGES_REQUESTED** [review]
LGTM overall. Waiting for some clarifications before we merge this.

File: bamboo-hr.sgnl.yaml
Comment (arooxa) [reply bamboo-hr.sgnl.yaml:43]: I just tested this in Postman on the SoR and it works without setting a password, but unfortunately the adapter has validation to make sure the password is non-empty. There also might be validation in the console/ingestion service to make sure that this field is non-empty.

File: bamboo-hr.sgnl.yaml
Comment (marc-sgnl) [reply bamboo-hr.sgnl.yaml:37]: Candidly, I really like this level of content here. It allows us to take it and reference in help.sgnl.ai in addition to what's in the template.

File: bamboo-hr.sgnl.yaml
Comment (marc-sgnl) [reply bamboo-hr.sgnl.yaml:96]: `Manager` SGTM here so it's consistent (similar to Group and others)

File: bamboo-hr.sgnl.yaml
Comment (r-rakshith) [reply bamboo-hr.sgnl.yaml:37]: My concern is these super helpful comments going unnoticed. 

A typical admin will see this, click on the SoR and land in SoR config page. If there's confusion in adapter-config, my instinct would be to check help.sgnl.ai instead of the GH repo. Plus, the GH repo is something the admin does not directly interact with at all throughout the Console usage lifecycle.


image


As an improvement, we could 
- add a key called `docs` into catalog and put these comments as value
- make the FE show these docs when user hovers over a `?`/[info icon](https://www.iconfinder.com/icons/352432/info_icon)

WDYT @marc-sgnl?

File: bamboo-hr.sgnl.yaml
Comment (arooxa) [reply bamboo-hr.sgnl.yaml:2]: fixed

File: bamboo-hr.sgnl.yaml
Comment (arooxa) [reply bamboo-hr.sgnl.yaml:10]: fixed

File: bamboo-hr.sgnl.yaml
Comment (arooxa) [reply bamboo-hr.sgnl.yaml]: fixed

File: bamboo-hr.sgnl.yaml
Comment (arooxa) [reply bamboo-hr.sgnl.yaml]: fixed

File: bamboo-hr.sgnl.yaml
Comment (arooxa) [reply bamboo-hr.sgnl.yaml:74]: I personally like the attributeAlias since we aren't indexing into an entity's fields when defining the attribute. This relationship is very simple, but the last 3-4 catalog's I've created have used this attributeAlias field so I've been trying to standardize. WDYT?

Comment (r-rakshith) [reply bamboo-hr.sgnl.yaml:74]: Standardizing SGTM

File: bamboo-hr.sgnl.yaml
Comment (marc-sgnl) [reply bamboo-hr.sgnl.yaml]: I'm ok to pull them out of the template, but I would like to see all supported filters as as well as all possible adapter config values documented to ensure they are used appropriately. If we can achieve that, I'm ok to forego the rest of this content. It's incredibly useful in building the help pages, so I'd like to see the time the team spent researching contributing to making our docs better.

Comment (arooxa) [reply bamboo-hr.sgnl.yaml]: Let's keep it in the templates for now, and we can always move them or duplicate it in other places afterwards. WDYT @r-rakshith ?

**Review by r-rakshith — APPROVED** [review]

File: bamboo-hr.sgnl.yaml
Comment (r-rakshith) [reply bamboo-hr.sgnl.yaml]: https://app.shortcut.com/sgnl/story/25002

Adding a feature request here for more dynamic and better documentation.

**Pull request merged**
